src/main/java/com/google/sps/data/QuestionDatabase.java

```diff
@@ -0,0 +1,66 @@
+package com.google.sps.data;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.Query;
+import java.util.ArrayList;
+import java.util.List;
+
+/* abstract quiz question database, to put and query for QuestionAndChoices */
+public abstract class QuestionDatabase<T extends QuestionChoice> {
+  private DatastoreService datastore;
+  private ArrayList<QuestionAndChoices<T>> questionAndChoices;
+  private static String entityQueryString;
+  private static Query query;
+  public static final String QUESTION_QUERY_STRING = "question";
+  public static final String CHOICETEXT_QUERY_STRING = "choicetext";
```

Comment:
This probably needs to change the query name altogether, but camel case "choiceText" naming convention

Reply:
+1

Reply:
Changed.